library/src/main/kotlin/io/intrepid/bleidiom/BleIdiomImpl.kt

```diff
@@ -203,26 +207,25 @@ internal class BleCharValueDelegate<Val : Any>() : BleCharHandlerDSL<Val> {
             letMany(uuid, service?.sharedConnection, toByteArray) { charUUID, connection, transform ->
                 connection.firstOrError().flatMapTry {
                     val bytes = transform(value)
-                    val writeObs = if (bytes.size > mtu) {
+                    val writeObs : Observable<ByteArray> = if (bytes.size > mtu) {
```

Comment:
This should be a `Single<ByteArray>` and then the `toObservable()` can be removed from line 223.
The `build()` method of the LongWriteBuilder seems to return an Observable, not a Single. This decision by RxAndroidBLE looks odd to me. Still, call `singleOrError` on that instead: `.build().singleOrError()`